src/main/java/com/synopsys/integration/detect/configuration/validation/DetectConfigurationValidator.java

```diff
@@ -0,0 +1,123 @@
+ * synopsys-detect
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detect.configuration.validation;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.configuration.config.PropertyConfiguration;
+import com.synopsys.integration.configuration.help.PropertyConfigurationHelpContext;
+import com.synopsys.integration.configuration.property.Property;
+import com.synopsys.integration.configuration.property.PropertyDeprecationInfo;
+import com.synopsys.integration.detect.configuration.DetectInfo;
+import com.synopsys.integration.detect.configuration.DetectProperties;
+import com.synopsys.integration.detect.configuration.DetectUserFriendlyException;
+import com.synopsys.integration.detect.configuration.enumeration.ExitCodeType;
+import com.synopsys.integration.detect.lifecycle.boot.DetectBootResult;
+import com.synopsys.integration.detect.lifecycle.shutdown.ExitCodeRequest;
+import com.synopsys.integration.detect.workflow.event.Event;
+import com.synopsys.integration.detect.workflow.event.EventSystem;
+import com.synopsys.integration.detect.workflow.report.writer.InfoLogReportWriter;
+import com.synopsys.integration.detect.workflow.status.DetectIssue;
+import com.synopsys.integration.detect.workflow.status.DetectIssueType;
+
+public class DetectConfigurationValidator {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final EventSystem eventSystem;
+    private final DetectInfo detectInfo;
+    private final InfoLogReportWriter infoLogReportWriter;
+
+    public DetectConfigurationValidator(EventSystem eventSystem, DetectInfo detectInfo) {
+        this.eventSystem = eventSystem;
+        this.detectInfo = detectInfo;
+        this.infoLogReportWriter = new InfoLogReportWriter();
+    }
+
+    public DetectConfigurationState processDetectConfiguration(PropertyConfiguration detectConfiguration) throws IllegalAccessException {
```

Comment:
I think process configuration is a little vague, as well as ConfigurationState. I don't really know what they do without looking into it. I am also not entirely sure why it contains a DetectConfiguration. Perhaps "checkForDeprecations" and DeprecationResult would be more fitting.